File: Medium/2025-11-23-1262-Greatest-Sum-Divisible-by-Three/solution.js
var maxSumDivThree = function(nums) {
    let sum = 0;
    let mod = [0, Infinity, Infinity];
    
    for (let num of nums) {
        sum += num;
        let newMod = [...mod];
        
        for (let i = 0; i < 3; i++) {
            newMod[(i + num) % 3] = Math.min(newMod[(i + num) % 3], mod[i] + num);
        }
        
        mod = newMod;
    }
    
    return sum - mod[1] === Infinity ? sum : sum - mod[1];
};

File: Medium/2025-11-23-1262-Greatest-Sum-Divisible-by-Three/solution.py
class Solution:
    def maxSumDivThree(self, nums: List[int]) -> int:
        total_sum = sum(nums)
        remainder = total_sum % 3
        
        if remainder == 0:
            return total_sum
        
        min1, min2 = float('inf'), float('inf')
        
        for num in nums:
            if num % 3 == remainder:
                min1 = min(min1, num)
            elif num % 3 == (remainder + 1) % 3:
                min2 = min(min2, num)
        
        if remainder == 1:
            if min1 != float('inf'):
                return total_sum - min1
            if min2 != float('inf'):
                return total_sum - min1 - min2 * 2
        else:
            if min2 != float('inf'):
                return total_sum - min2
            if min1 != float('inf'):
                return total_sum - min1 * 2 - min2
        
        return 0